src/Compilers/Core/Portable/Syntax/SyntaxDiffer.cs

```diff
@@ -568,6 +582,37 @@ private void RecordChange(ChangeRecord change)
             _changes.Add(change);
         }
 
+        private void RecordChange(TextChangeRange textChangeRange, SyntaxNodeOrToken removedNode, SyntaxNodeOrToken insertedNode)
+        {
+            if (_changes.Count > 0)
+            {
+                var last = _changes[_changes.Count - 1];
+                if (last.Range.Span.End == textChangeRange.Span.Start)
+                {
+                    // merge changes...
+                    last.OldNodes?.Enqueue(removedNode);
+                    last.NewNodes?.Enqueue(insertedNode);
+                    _changes[_changes.Count - 1] = new ChangeRecord(
+                        new TextChangeRange(new TextSpan(last.Range.Span.Start, last.Range.Span.Length + textChangeRange.Span.Length), last.Range.NewLength + textChangeRange.NewLength),
```

Comment:
would this be any clearer/cleaner as "TextSpan.FromBounds(last.Range.Span.Start, textChangeRange.Span.End)"?  (if that is even correct of course).